packages/regl-worldview/src/commands/GLText.js

```diff
@@ -354,7 +374,10 @@ function makeTextCommand(alphabet?: string[]) {
         let y = 0;
         let markerInstances = 0;
 
-        const fgColor = marker.colors?.[0] || marker.color || BG_COLOR_LIGHT;
+        // If we need to render text for hitmap framebuffer, we only render the polygons using
+        // the foreground color (which needs to be converted to RGBA since it's a vec4).
+        // See comment on fragment shader above
+        const fgColor = forHitmap ? hitmapColor(marker.color) : marker.colors?.[0] || marker.color || BG_COLOR_LIGHT;
```

Comment:
Since we use the color to locate objectId, would be good to be more strict about fallbacks. 

When `hitmap` is enabled, the `color` or `colors` should always be present.  
 since `GLText` doesn't have any `colors` as input props, we don't need to fall back to `marker.colors?.[0]`? https://github.com/cruise-automation/webviz/blob/master/packages/regl-worldview/src/utils/getChildrenForHitmapDefaults.js#L75-L104

If `color` is not present, we could fallback to `[0, 0, 0, 1]` which is the default canvas color, similar to GLTFScene: https://github.com/cruise-automation/webviz/blob/master/packages/regl-worldview/src/commands/GLTFScene.js#L203

Reply:
@vidaaudrey Sorry, I'm not sure I understand. Should I fallback to `marker.colors?.[0]` if `marker.color` is undefined when rendering for the hitmap? And only then fallback to black if both `marker.color` and `marker.colors?.[0]` are both undefined? 

Also, I'm already falling back to black in `hitmapColor()` if marker.color is not valid in order to avoid selection. Is that correct?

Reply:
Hi sorry, i didn't see your fallback inlined `: markerColor?.[0] ?? 0,`.
nit: would be nice to write it like `const fgColor = isHitmap ? (marker.color || [0, 0, 0, 1]) : ...` so it's a more explicit.  Separately, could also fix the polygon to support vec4 as color input if it doesn't already.

Reply:
On that note maybe we already have a utility helper somewhere to convert between object/array representations?

Reply:
yeah, quite a few utils here: https://github.com/cruise-automation/webviz/blob/master/packages/regl-worldview/src/utils/commandUtils.js#L55

Reply:
@vidaaudrey @janpaul123 Check out my latest commit. I modified TextMarker to use `Color | Vec4` for both `color` and `colors` properties, in the same way other commands are doing it. That let me use the existing `vec4ToRGBA` utils but I had to add a new one called `toColor` for color conversion in order to fix flow errors because of the union type. Please take a look and let me know what you think.

Reply:
Yeah looks good to me!